lib/net.js

```diff
@@ -899,24 +899,18 @@ function connect(self, address, port, addressType, localAddress, localPort) {
 }
 
 
-Socket.prototype.connect = function(options, cb) {
+Socket.prototype.connect = function() {
+  const args = new Array(arguments.length);
+  for (var i = 0; i < arguments.length; i++)
+    args[i] = arguments[i];
+  // TODO(joyeecheung): use destructuring when V8 is fast enough
```

Comment:
I'll be OK with that.

The timeouts are added in https://github.com/nodejs/node/pull/8101/files, I can't see any harm adding them to `Socket.prototype.connect`, so +1 to reduce code duplicate.